### PR TITLE
Add support trace on MPS backend

### LINKED ISSUE #87221: [MPS] Add support for aten::trace for MPS backend
### 🐛 Describe the bug

First time contributors are welcome! 🙂
Add support for [aten::trace](https://pytorch.org/docs/stable/generated/torch.trace.html) for MPS backend. Generic support for adding operations to MPS backend is captured here: https://github.com/pytorch/pytorch/wiki/MPS-Backend#adding-op-for-mps-backend

### Versions

N/A

cc @kulinseth @albanD @malfet @razarmehr @abhudev

### PR DESCRIPTION
**Pull request by pumetu**
Fixes [#87221](https://github.com/pytorch/pytorch/issues/87221)
`trace` now supported on MPS

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/TriangularOps.mm [hunk 1]
```diff
@@ -172,5 +172,82 @@
...
+    self.scalar_type(),
+    c10::nullopt,
+    kMPS,
+    c10::nullopt,
+    c10::nullopt
+  );
+
+  @autoreleasepool {
+    string key = "trace_mps_out" + mps::getTensorsStringKey({self});
+    CachedGraph* cachedGraph = static_cast<CachedGraph *>(cache_->LookUp(key));
```
Comment (malfet) [inline aten/src/ATen/native/mps/operations/TriangularOps.mm:208]: Nit
```suggestion
    CachedGraph* cachedGraph = cache_->LookUpAs<CachedGraph>(key);
```

File: aten/src/ATen/native/mps/operations/TriangularOps.mm [hunk 2]
```diff
@@ -172,5 +172,82 @@
...
+  using namespace mps;
+  MPSStream* stream = getCurrentMPSStream();
+
+  // Derive from MPSCachedGraph
+  struct CachedGraph : public MPSCachedGraph
+  {
+    CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
+    MPSGraphTensor *inputTensor_ = nil;
+    MPSGraphTensor *outputTensor_ = nil;
+  };
```
Comment (malfet) [inline aten/src/ATen/native/mps/operations/TriangularOps.mm:191]: This is not needed, use https://github.com/pytorch/pytorch/blob/2e48b478e06b38a7468832d980d214441855547e/aten/src/ATen/native/mps/OperationUtils.h#L139-L144
```suggestion
```

**Comment by PumeTu** [comment]
I added new comments as suggested by the reviewer to switch from `CachedGraph` to `MPSUnaryCachedGraph`. Also, would I be able to get some guidance on how to fix the EasyCLA authorization, I haved signed the Contributor document and the error still seems to persist.

**Review by kulinseth — APPROVED** [review]
Looks good. Thanks.

**Comment by PumeTu** [comment]
I moved the `trace_mps` function to `ReduceOps.mm` because the output shape was incorrect in `TriangularOps.mm`, and the comments on the issue also recommended it.

**Review by malfet — APPROVED** [review]

**Comment by malfet** [comment]
@PumeTu please add an email you've authored this PR with to your github registered emails, and then easy CLA should pass

**Comment by PumeTu** [comment]
@malfet I was able to fix the CLA check by ammending the first commit to add my email in. All the commits were updated but the content should still be the same and all the functionalities should still work.

**Pull request closed without merging**